plan/plan_loader_test.go

```diff
@@ -65,3 +66,65 @@ func TestLoadOneFile(t *testing.T) {
 	}
 	assert.Equal(t, gotPlans, expectPlans)
 }
+
+func TestPlan_ShouldReturnDeepCopyOfPlan(t *testing.T) {
+	pl := NewLoader(
+		func(blip.ConfigPlans) ([]blip.Plan, error) { return []blip.Plan{}, nil },
+	)
+	planName := "foobar"
+	pl.sharedPlans = []planMeta{
+		{
+			name: planName,
+			plan: blip.Plan{
+				Levels: map[string]blip.Level{
+					"l1": {
+						Name: "l1",
+						Freq: "1s",
+						Collect: map[string]blip.Domain{
+							"d1": {
+								Name:    "d1",
+								Metrics: []string{"m1"},
+							},
+						},
+					},
+					"l2": {
+						Name: "l2",
+						Freq: "5s",
+						Collect: map[string]blip.Domain{
+							"d1": {
+								Name:    "d1",
+								Metrics: []string{"m2"},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	got, err := pl.Plan("", planName, nil)
+	require.NoError(t, err)
+
+	expected := pl.sharedPlans[0].plan
+	assert.Equal(t, expected, got)
+	verifyDeepCopy(t, expected, got)
+}
+
+// verifyDeepCopy verifies is a is a deep copy of b by comparing address of the slices and maps in blip.Plan.
+// This solution is not ideal, but there really isn't a good way to compare reference object addresses in go.
+// The converse is to attempt to mutate the slices and maps embedded within blip.Plan, but even more verbose.
+func verifyDeepCopy(t *testing.T, a, b blip.Plan) {
```

Comment:
I think it's best to avoid testing in non-test functions like this one because it makes test cases difficult to follow and modify. It's best when each `func TestX` case is truly isolated because then future engineers can understand properly functionality case by case (and move/change them independently, too).

Reply:
Sure. I'll remove this abstraction.